core/src/main/scala/scalafix/rewrite/SemanticApi.scala

```diff
@@ -13,17 +19,21 @@ import scala.meta._
   *
   * See [[ExplicitImplicit]] for an example usage of this semantic api.
   */
-trait SemanticApi {
+trait SemanticApi extends Mirror {
```

Comment:
How about we rename `SemanticApi` into `Mirror` (or `ScalafixMirror`) in order not to confuse people with multiple way of saying the same thing?